### PR TITLE
Refactor legal archive. Upgrade to latest Eve.

### PR DESCRIPTION
**Pull request by amagdas**
Replaces https://github.com/superdesk/superdesk/pull/682

### CODE REVIEW

**Comment by plamut** [comment]
@amagdas So basically everything is the same as in #682 , except for the bumped Eve version? So no need to review it again?

**Comment by sivakuna-aap** [comment]
@plamut Not exactly. Apart from bumping up eve version, he also refactored component and model of legal archive to use multi-db support.

**Comment by plamut** [comment]
OK, thanks, I'll have a closer look at it then (when I manage at least...).

File: server/apps/archive/commands.py [hunk 1]
```diff
@@ -28,26 +28,29 @@ def remove_expired_content(self):
         logger.info('Removing expired content')
         now = date_to_str(utcnow())
         items = self.get_expired_items(now)
+
         while items.count() > 0:
             for item in items:
                 logger.info('deleting {} expiry: {} now:{}'.format(item['_id'], item['expiry'], now))
                 superdesk.get_resource_service('archive').delete_action({'_id': str(item['_id'])})
```
Comment (mdhaman) [inline server/apps/archive/commands.py]: Are we checking if the spiked item is part of a package? Are we removing the referenced from the package?

**Comment by marwoodandrew** [comment]
:+1:

**Comment by mdhaman** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

**Pull request merged**